### DISCUSSION TITLE
Where do I go for help with...? Figure

### DISCUSSION THREAD

**Discussion by JessicaS11** [pydata/xarray#8990] [Q&A] [answered]
With so many great resources, it can be hard to know the best place to go for getting help. During a recent office hours, we brainstormed creating a "Where do I go for help with...?" figure modeled after one created a few years ago to direct hackweek participants to the right spot depending on their question. I've drafted the figure 
[SupportDecisionTree.pdf](https://github.com/pydata/xarray/files/15170942/SupportDecisionTree.pdf) and am now looking for feedback, especially:
* what's missing?
* where should we share it (I'm thinking [here](https://tutorial.xarray.dev/intro.html) and [here](https://docs.xarray.dev/en/stable/getting-started-guide/faq.html#) are options)?
* considerations, if any, for its ultimate version-controlled home (it's currently a 41KB .svg that I exported to pdf because that was the only export option from Inkscape that preserved the links)?
* any other constructive feedback, especially around accessibility.

I HAVE NOT done a detailed pass over for perfect alignment, etc., because that always changes anyway. I was also considering adding the Xarray logo in the top left, but didn't want it to get redundant since it'll be on pages that already have the logo.

@scottyhq @e-marshall

**Comment by dcherian** [comment]
Very nice work!

I'd add https://docs.xarray.dev/en/stable/contributing.html and remove the "dev meeting". We want to keep away usage questions from that meeting and leave it for more complex questions (e.g. building out new integrations / features).

>  considerations, if any, for its ultimate version-controlled home 

I wonder if we can build it out in HTML. (cc @andersy005)

**Reply by benbovy** [reply]
> I wonder if we can build it out in HTML

If we want to include it in the docs there is https://github.com/mgaitan/sphinxcontrib-mermaid.

**Reply by JessicaS11** [reply]
> I'd add https://docs.xarray.dev/en/stable/contributing.html and remove the "dev meeting". We want to keep away usage questions from that meeting and leave it for more complex questions (e.g. building out new integrations / features).

Agreed. However, my actually putting this together was in part motivated by the fact that we've gotten a couple cases where folks who were working on contributions (e.g. open PRs) were asking for dev feedback at office hours in order to complete their contributions. They didn't know that they could go to the dev meeting or that it was the best place for their questions. How about modifying the text to something like "Xarray Developer meeting* - all are welcome" and "*Please note developer meetings are short check-ins and cannot address usage questions" to make that clear?

I will add the contributors guide - thanks for catching that was missing!

> If we want to include it in the docs there is https://github.com/mgaitan/sphinxcontrib-mermaid.

I actually started out playing with mermaid for ease of including in the docs, but quickly ran up against formatting issues trying to have variable length boxes and blank spaces (a few were noted in open issues). If someone is more of a mermaid wizard than me though, I like the idea!

**Comment by TomNicholas** [comment]
I like this idea!

This might be pedantic but this isn't a decision "tree" 😅 From your description I imagined a flow chart more like this (though this technically isn't a tree either!)

![image](https://github.com/pydata/xarray/assets/35968931/0c5424cf-bcb3-46af-a60a-8e6b1cf786b5)

Would this type of structure be more useful?

**Reply by JessicaS11** [reply]
> this isn't a decision "tree"

Pedantic and true! At least I didn't just name the file "xarray figure". 😅 I'll rename for the next round.

> Would this type of structure be more useful?

I'm having trouble imagining it not getting too spiderweb-y, but I like the idea of having a logic-based flow. I'll keep playing with this idea and am happy to see suggestions for how it could work. Graphics don't come naturally to me.

**Reply by JessicaS11** [reply]
@TomNicholas It's still a decidedly *rough* draft, but:

<img width="1015" alt="image" src="https://github.com/pydata/xarray/assets/11756442/98cdfa89-8bef-4e09-8e8d-8d8ddea008d2">

I'm creating it in [Mermaid.live](https://mermaid.live/edit#pako:eNp9VNtu2kAQ_ZXNvoRIgYSaq1U1akuV8pAWkUitinlYvGOzxXjdvXBpnH_vrG0cSKIaCfDOnNkzc87uIw0lB-rTZrMZpEaYBHzyjW1EzIxIY_KTKcX2ZApaWhWCDtIi0X2iRG7DJVOGPIyClOAjUqPk7AckoVwDMbJCn5GvcktClpItkCUk2c28zLeaxdCYVXsYa6QSLNFl0D1VhMvwaPFWyjiBq3vDwhX5skMKaQzzizLh42RcFzyFwc5AqoVMz3UROSAcoRoyEjq02mUdISO28CPW5BiTipOjVHx9RfZ7FIkQsGWrjmpMHElZgjBQ813YuDGbQiZxjCzlZKJkJrUbkwL_FYVYmKVd1GP5D9nTTKG1hbrjUKJOYlE3_bl8d-PH2dxawWt6mWrMAnpac2KTBA3xBysa0phMLwJ6SOewacxGsIFEZuCK3QE4Gx3iqmh09oCSrTSJpCIJmMJnVpNVKrdn89JbtZtIs_khd_bhkowxC6qWbvLSPUdGKlIDukWWz2Jr0kjECsiI6RUSzZ1BSsxL5PickyJVO_OupSq9mhffLxmNkbxFuZgTkGAfZp_J3L2UmdouYsWypVupxj37ZGNdaFzP2_GbP0tXo49EOom-P-GJ5ysSuyvGORHmLH8TUZH9bVGqLUsNcAczgArupc0rQUoIpPwwmdqbDr7ekyxxh5ch_SxTcgMct8vUW1KNyVbo5cF36PWEt1qt_GgOB0ymSFV9MkWDkj9WgMmdhZ7r0ku6BrVmguMN9ejWAmqWsIaA-viXQ8RsYgIapE-YytDB9_s0pL5RFi6pzTgzMBIMpVhTdHCicTVjKfUf6Y76nW6r1xsMB16nO-z03_W6vUu6p36777W8zvVwMGh3h-3-sP90Sf9KiRXarWuv53nDYd8bdPDX6xTlfhXBck_gAo_RXXmnFlfr0z9E5b50) so we could put it directly into a JupyterNotebook using the [`%%mermaid_magic`](https://opencomputinglab.github.io/SubjectMatterNotebooks/diagram/js-diagramming-magics.html). If the link doesn't include the flowchart code let me know and I'll share here.

👍 to proceed with building out this Mermaid flowchart version
👎 if I should keep going with the help diagram figure I originally posted

**Reply by TomNicholas** [reply]
Oh I really like this! +1 from me (but others should weigh in)

One nit is I think remove xarray discussions as a recommendation for bugs / feature proposals, as those should all be issues. Discussions are mostly for usage questions that weren't answered by the docs.

**Reply by TomNicholas** [reply]
I think also there could be more routes to the developers meeting, such as "I wish xarray could..." -> "raise an issue to discuss" -> "my issue is quiet" -> "dev meeting"

**Reply by JessicaS11** [reply]
[Next iteration!](https://mermaid.live/edit#pako:eNqNVW1v2kgQ_iub7YdSCdPEFyBYVau0QC_StYeSSnc6nA-LPTZ7LLvuvgRoyH-_Wb_wEvhwtgT27vPMzM48M36miUqBRjQIglhabgVE5Dt74jmzXObkb6Y125B7MMrpBEwsS6C_M6FWyZxpS34MY0nw4tJqNf0LRKKWQKyq2Rfkd7UiCZNkBWQOovj0GEXRTwfmO_rOlrZiO8NyaE0zFmUsQMNiMVNMp4EzoJs4rLNKcyZMRfFXxmaeoIGl6LTGpSo5heRK5QLI1_Lvfb1oLEsWgXoC7c9DHvwrGa3Rv8zh8R0GyuRxnLeTOx_l__VaQ2BtQRqu5FtT4s6a9snZ2c65nbtZY3vITeKMN3DqIsU9pdNDKL4ewjzK59E0mD-zjCeAhXHavAbmQs2ATHwCVGUMQedzMXN5a3oPhUIVMJmSiVaFMr7KGqLTAhydiBuDEjhrNlGoJD5rxDBTCktUgGyoX6p9LwZM51fH0_PhFbo1jemx94kTAvXs5WdJa3L_LqZnuSk8Nf6rzA3hCQRG4V1-A_DtcZaoy3RMf6CCFoZkShMBtmwmZ8hCqtXFifz9vWsgEgQft75jUkXukNPI69O2apGDbimhMV3hyfYCM6Ql-ALIkJkFHm7rBVtxXjPv3qakhBrfr0ulq_bclr-vI7rDozgsMfNFJ3gquynU1r9USONmuWbF3K_U9Zt-drkpdbErmI_vcS-MHfug6ke7tet_HRZrxaSF1M8VC1jDjXLbOtlHlA9HR0N0xtfEzsHvXWzPeikpyw0phB9SDGMuCo0zId3jG_SeVz2BTA-Te5q1FTfzRrfYSSLtdDp12hqRktr95B6bgvx0HOzW62_XYw2gbJlzmCZIxBW6MZwIZswQMnKgNUyGENGbwXA0Gl-3DYa5gOhNeNn7cjM44e2VXdPG49vby-6ONvqtP77qn9Bman1IGXdHwxPKjnAgl5rY2BNcLh7sBic2hsCcQIuICwz_BVF4Wdh2ooTS--A9h7bpEvSS8RQ_ac9-LaZY-iXENMLH2lBM29WWgcRpbjd_-NauIELh_IppLF_QFsMJ87CRCY2sdtCmrkiZhSFnqPQlxdEgDK4WTNLoma5pdN0ZXA4G_d6gOwjDm-5Vr003NLrq9TphGF53-zc33f51GL606S-l0MBVSf6nfK48QMpxqH2rvsnlp7lNtXL5vPb28h_DZIWl)

![image](https://github.com/pydata/xarray/assets/11756442/6b30b159-4302-4845-80fa-58f79b356605)

I'm leaning towards putting this in the documentation (and then linking to it from the tutorials). I'll open a PR once we [debug including a mermaid diagram in a md file](https://github.com/xarray-contrib/xarray-tutorial/pull/266).

**Reply by TomNicholas** [reply]
This is absolutely great @JessicaS11 !

Another (totally optional) nit: most Pull Requests are resolved without having to attend a developers meeting - perhaps there could be another blue end point square box like "PR approved and merged!"

**Reply by dcherian** [reply]
"Gentle pings" are nice low-effort ways to get attention to open PRs and issues.

**Comment by JessicaS11** [comment, marked as answer]
https://docs.xarray.dev/en/latest/help-diagram.html

**Reply by TomNicholas** [reply]
Amazing! We should tweet about this :)
